velox/connectors/hive/storage_adapters/hdfs/HdfsWriteFile.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "HdfsWriteFile.h"
+#include <hdfs/OutputStream.h>
+#include <hdfs/hdfs.h>
+#include <iostream>
+#include <regex>
+
+namespace facebook::velox {
+
+HdfsWriteFile::HdfsWriteFile(std::string& path) {
+  struct hdfsBuilder* builder = hdfsNewBuilder();
+  // read hdfs client conf from hdfs-client.xml from LIBHDFS3_CONF
+  hdfsBuilderSetNameNode(builder, "default");
+  hdfsClient_ = hdfsBuilderConnect(builder);
+  VELOX_CHECK(
+      hdfsClient_ != nullptr,
+      "failed to get hdfs client with default setting, with error:" +
```

Comment:
- use formatting strings to avoid constructing the string until the error has occurred
- use VELOX_CHECK_NOT_NULL
- use full sentences for error message and remove unnecessary text (with error)

```
VELOX_CHECK_NOT_NULL(
      hdfsClient_,
      "Failed to create HDFS client with default setting: {}", 
     std::string(hdfsGetLastError()))
```